mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java

```diff
@@ -256,82 +256,72 @@
     }
 
-    @ApiOperation(value = "Adds or updates an association between a mico application and a mico service. An existing and" +
-        " already associated mico service with an equal shortName it will be replaced." +
-        " Only one mico service in one specific version is allowed per mico application.")
+    @ApiOperation(value = "Adds or updates an association between a MicoApplication and a MicoService. An existing and" +
+        " already associated MicoService with an equal short name will be replaced with its new version." +
+        " Only one MicoService in one specific version is allowed per MicoApplication.")
     @PostMapping("/{" + PATH_VARIABLE_SHORT_NAME + "}/{" + PATH_VARIABLE_VERSION + "}/" + PATH_SERVICES + "/{" + PATH_VARIABLE_SERVICE_SHORT_NAME + "}/{" + PATH_VARIABLE_SERVICE_VERSION + "}")
-    public ResponseEntity<Void> addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String micoApplicationShortName,
-                                                        @PathVariable(PATH_VARIABLE_VERSION) String micoApplicationVersion,
-                                                        @PathVariable(PATH_VARIABLE_SERVICE_SHORT_NAME) String micoServiceShortName,
-                                                        @PathVariable(PATH_VARIABLE_SERVICE_VERSION) String micoServiceVersion) {
-        MicoApplication micoApplication = getApplicationFromDatabase(micoApplicationShortName, micoApplicationVersion);
-        MicoService micoServiceFromParameter = getServiceFromDatabase(micoServiceShortName, micoServiceVersion);
-        List<MicoService> micoServicesWithIdenticalShortName = getAllMicoServicesFromMicoApplicationWithShortName(micoServiceShortName, micoApplication);
-        if (micoServicesWithIdenticalShortName.size() > MAX_MICO_SERVICES_WITH_SAME_SHORT_NAME) {
+    public ResponseEntity<Void> addServiceToApplication(@PathVariable(PATH_VARIABLE_SHORT_NAME) String applicationShortName,
+                                                        @PathVariable(PATH_VARIABLE_VERSION) String applicationVersion,
+                                                        @PathVariable(PATH_VARIABLE_SERVICE_SHORT_NAME) String serviceShortName,
+                                                        @PathVariable(PATH_VARIABLE_SERVICE_VERSION) String serviceVersion) {
+    	// Retrieve application and service from database (checks whether they exist)
+        MicoApplication application = getApplicationFromDatabase(applicationShortName, applicationVersion);
+        MicoService service = getServiceFromDatabase(serviceShortName, serviceVersion);
+        
+        // Find all services with identical short name within this application
+		List<MicoService> micoServicesWithIdenticalShortName = serviceRepository
+		    .findAllByApplicationAndShortName(applicationShortName, applicationVersion, serviceShortName);
```

Comment:
Why did you change this? It is not necessary to query the information from the db again.  Everything needed is already there. We now get the same information twice from the db.